src/test/java/test/SearchPageTests.java

```diff
@@ -0,0 +1,71 @@
+package test;
+
+import core.AllureUtils;
+import core.WebDriverSingleton;
+import org.junit.*;
+import org.junit.rules.TestRule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.openqa.selenium.WebDriver;
+import pages.BasePage;
+import pages.HeaderBlock;
+import pages.SearchResultPage;
+
+import java.text.ParseException;
+import java.util.Locale;
+
+public class SearchPageTests {
+
+    private WebDriver driver;
+    private SearchResultPage searchResultPage = new SearchResultPage();
+    private HeaderBlock header = new HeaderBlock();
+
+    @Rule
+    public TestRule watchman = new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+           AllureUtils.makeScreenshot();
+        }
+
+        @Override
+        protected void starting(Description description) {
+            driver = WebDriverSingleton.getWebDriver();
+            driver.get(BasePage.BASE_PAGE_URL);
+        }
+
+        @Override
+        protected void finished(Description description) {
+            WebDriverSingleton.shutDownWebDriver();
+        }
+    };
+
+    @Test
+    public void first10TitlesContainSearchCriteriaTest() {
+        header.searchFor("iphone");
+        for (int i = 0; i < 10; i++) {
+            Assert.assertTrue(String.format("Title %s should contain iphone", searchResultPage.getTitles().get(i))
```

Comment:
`searchResultPage.getTitles().get(i)` can be extracted to loop variable here and in below tests

Reply:
done here and below